File: csharp/O/After/Strategy/BreakerDCStrategy.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace After.Strategy;

public class BreakerDCStrategy : IProtectionStrategy
{
    public double CalcBreakingCapacity()
    {
        return 3;
    }

    public double CalcIn()
    {
        return 3;
    }

    public bool CheckProtection()
    {
        return true;
    }
}

File: js-ts/O/after/electrical-protection/Strategy/BreakerDCStrategy.ts
import { IProtectionStrategy } from "./IProtectionStrategy.ts";

export class BreakerDCStrategy implements IProtectionStrategy {
    public calcIn(In: number): number {
        return In + 2;
    }

    public calcBreakingCapacity(): number {
        return 2;
    }

    public checkProtection(): boolean {
        return true;
    }
}
